src/Rules/Variables/UnsetRule.php

```diff
@@ -103,6 +105,36 @@ private function canBeUnset(Node $node, Scope $scope): ?IdentifierRuleError
 					->identifier($propertyReflection->isReadOnly() ? 'unset.readOnlyProperty' : 'unset.readOnlyPropertyByPhpDoc')
 					->build();
 			}
+
+			if ($propertyReflection->isHooked()) {
+				return RuleErrorBuilder::message(
+					sprintf(
+						'Cannot unset hooked %s::$%s property.',
+						$propertyReflection->getDeclaringClass()->getDisplayName(),
+						$foundPropertyReflection->getName(),
+					),
+				)
+					->line($node->getStartLine())
+					->identifier('unset.hookedProperty')
+					->build();
+			} elseif ($this->phpVersion->supportsPropertyHooks()) {
+				if (
+					!$propertyReflection->isPrivate()
+					&& !$propertyReflection->isFinal()->yes()
+					&& !$propertyReflection->getDeclaringClass()->isFinal()
+				) {
```

Comment:
I also though about adding 

```
public function isHookedInSubclass(): TrinaryLogic {
		if (
			!$this->isPrivate()
			&& !$this->isFinal()->yes()
			&& !$this->getDeclaringClass()->isFinal()
		) {
			return TrinaryLogic::createMaybe();
		}

		return TrinaryLogic::createNo();
	}
```

to `PhpPropertyReflection`. wdyt?

Reply:
Not needed right now :) I'd have to this about that more.